src/bundles/gateway.js

```diff
@@ -150,36 +253,72 @@ async function checkViaImgUrl (imgUrl) {
  * Checks if a given gateway URL is functioning correctly by verifying image loading and redirection.
  *
  * @param {string} gatewayUrl - The URL of the gateway to be checked.
- * @returns {Promise<boolean>} A promise that resolves to true if the gateway is functioning correctly, otherwise false.
+ * @returns {Promise<{isValid: boolean, securityWarnings: string[], errors: string[], securityScore?: number}>} A promise that resolves to an object with validation results including security warnings.
  */
 export async function checkSubdomainGateway (gatewayUrl) {
   if (gatewayUrl === DEFAULT_SUBDOMAIN_GATEWAY || gatewayUrl === TEST_SUBDOMAIN_GATEWAY) {
     // avoid sending probe requests to the default gateway every time Settings page is opened
     // also skip validation for test gateways
-    return true
+    return { isValid: true, securityWarnings: [], errors: [] }
+  }
+
+  // Perform security validation first
+  const securityValidation = validateSubdomainGatewaySecurity(gatewayUrl)
+  if (!securityValidation.isValid) {
+    return {
+      isValid: securityValidation.isValid,
+      securityWarnings: securityValidation.warnings,
+      errors: securityValidation.errors,
+      securityScore: securityValidation.securityScore
+    }
   }
   /** @type {URL} */
   let imgSubdomainUrl
   /** @type {URL} */
   let imgRedirectedPathUrl
   try {
     const gwUrl = new URL(gatewayUrl)
-    imgSubdomainUrl = new URL(`${gwUrl.protocol}//${IMG_HASH_1PX}.ipfs.${gwUrl.hostname}/?now=${Date.now()}&filename=1x1.png#x-ipfs-companion-no-redirect`)
+    // Check if the gateway URL already has an IPFS subdomain
+    // If hostname starts with 'ipfs.', we need to handle it differently
+    const hasIpfsSubdomain = gwUrl.hostname.startsWith('ipfs.')
+
+    if (hasIpfsSubdomain) {
```

Comment:
Can you not just ... not do this check?

Let the user add whatever subdomain or not he wants?